spec/globus/client_spec.rb

```diff
@@ -88,45 +83,68 @@
     end
   end
 
-  describe ".mkdir" do
-    it "invokes mkdir on the Endpoint class, injecting config" do
-      described_class.mkdir(user_id:, work_id:, work_version:)
-      expect(fake_endpoint).to have_received(:mkdir).once
+  describe "public class methods" do
+    let(:fake_instance) do
+      instance_double(
+        described_class,
+        mkdir: nil,
+        file_count: nil,
+        total_size: nil
+      )
     end
-  end
 
-  describe ".file_count" do
-    it "invokes file_count on the Endpoint class, injecting config" do
-      described_class.file_count(user_id:, work_id:, work_version:)
-      expect(fake_endpoint).to have_received(:file_count).once
+    before do
+      allow(described_class).to receive(:instance).and_return(fake_instance)
```

Comment:
I think this is what I was missing.

Reply:
np, overtesting the class methods was my own sin from earlier so I was eager to clear that up